.github/dockerfiles/ov_build/ubuntu_24_04_x64/Dockerfile

```diff
@@ -0,0 +1,68 @@
+FROM openvinogithubactions.azurecr.io/dockerhub/ubuntu:24.04
+
+USER root
+
+# APT configuration
+RUN echo 'Acquire::Retries "10";' > /etc/apt/apt.conf && \
+    echo 'APT::Get::Assume-Yes "true";' >> /etc/apt/apt.conf && \
+    echo 'APT::Get::Fix-Broken "true";' >> /etc/apt/apt.conf && \
+    echo 'APT::Get::no-install-recommends "true";' >> /etc/apt/apt.conf
+
+ENV DEBIAN_FRONTEND="noninteractive" \
+    TZ="Europe/London"
+
+RUN apt-get update && \
+    apt-get install software-properties-common && \
+    add-apt-repository --yes --no-update ppa:git-core/ppa && \
+    add-apt-repository --yes --no-update ppa:deadsnakes/ppa && \
+    apt-get update && \
+    apt-get install \
+        curl \
+        git \
+        gpg-agent \
+        tzdata \
+        # Python
+        python3-dev \
+        python3-venv \
+        python3-pip \
+        python3-full \
+        # For Java API
+        default-jdk \
+        # Compiler
+        gcc-10 \
+        g++-10 \
+        && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install build dependencies
+ADD install_build_dependencies.sh /install_build_dependencies.sh
+RUN chmod +x /install_build_dependencies.sh && \
+    /install_build_dependencies.sh && \
+    rm -rf /var/lib/apt/lists/*
+
+# Set gcc-10 as a default compiler
+RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 30 && \
+    update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-10 30
```

Comment:
what errors do you see w/o gcc-10?

Reply:
Protobuf/onnx do not build:
`2024-07-31T11:31:48.7858510Z       gmake[1]: *** [CMakeFiles/Makefile2:1210: _deps/protobuf-build/CMakeFiles/libprotobuf.dir/all] Error 2
2024-07-31T11:31:48.7858620Z       gmake: *** [Makefile:136: all] Error 2
2024-07-31T11:31:48.7858730Z       Traceback (most recent call last):
2024-07-31T11:31:48.7859190Z         File "/venv/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
2024-07-31T11:31:48.7859389Z           main()
2024-07-31T11:31:48.7859809Z         File "/venv/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
2024-07-31T11:31:48.7859978Z           json_out['return_val'] = hook(**hook_input['kwargs'])
2024-07-31T11:31:48.7860088Z                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-07-31T11:31:48.7860548Z         File "/venv/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
2024-07-31T11:31:48.7860769Z           return _build_backend().build_wheel(wheel_directory, config_settings,
2024-07-31T11:31:48.7860883Z                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-07-31T11:31:48.7861358Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 415, in build_wheel
2024-07-31T11:31:48.7861475Z           return self._build_with_temp_dir(
2024-07-31T11:31:48.7861567Z                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-07-31T11:31:48.7862086Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 397, in _build_with_temp_dir
2024-07-31T11:31:48.7862182Z           self.run_setup()
2024-07-31T11:31:48.7862628Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 313, in run_setup
2024-07-31T11:31:48.7862739Z           exec(code, locals())
2024-07-31T11:31:48.7862848Z         File "<string>", line 327, in <module>
2024-07-31T11:31:48.7863275Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/__init__.py", line 108, in setup
2024-07-31T11:31:48.7863399Z           return distutils.core.setup(**attrs)
2024-07-31T11:31:48.7863492Z                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-07-31T11:31:48.7863952Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 184, in setup
2024-07-31T11:31:48.7864058Z           return run_commands(dist)
2024-07-31T11:31:48.7864144Z                  ^^^^^^^^^^^^^^^^^^
2024-07-31T11:31:48.7864638Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 200, in run_commands
2024-07-31T11:31:48.7864734Z           dist.run_commands()
2024-07-31T11:31:48.7865217Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 970, in run_commands
2024-07-31T11:31:48.7865326Z           self.run_command(cmd)
2024-07-31T11:31:48.7865747Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/dist.py", line 945, in run_command
2024-07-31T11:31:48.7865870Z           super().run_command(command)
2024-07-31T11:31:48.7866344Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
2024-07-31T11:31:48.7866438Z           cmd_obj.run()
2024-07-31T11:31:48.7866916Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/command/bdist_wheel.py", line 373, in run
2024-07-31T11:31:48.7867016Z           self.run_command("build")
2024-07-31T11:31:48.7867574Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
2024-07-31T11:31:48.7867718Z           self.distribution.run_command(command)
2024-07-31T11:31:48.7868145Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/dist.py", line 945, in run_command
2024-07-31T11:31:48.7868265Z           super().run_command(command)
2024-07-31T11:31:48.7868736Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
2024-07-31T11:31:48.7868827Z           cmd_obj.run()
2024-07-31T11:31:48.7869404Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/_distutils/command/build.py", line 135, in run
2024-07-31T11:31:48.7869505Z           self.run_command(cmd_name)
2024-07-31T11:31:48.7869974Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
2024-07-31T11:31:48.7870116Z           self.distribution.run_command(command)
2024-07-31T11:31:48.7870542Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/dist.py", line 945, in run_command
2024-07-31T11:31:48.7870645Z           super().run_command(command)
2024-07-31T11:31:48.7871133Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
2024-07-31T11:31:48.7871221Z           cmd_obj.run()
2024-07-31T11:31:48.7871341Z         File "<string>", line 264, in run
2024-07-31T11:31:48.7871807Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
2024-07-31T11:31:48.7871936Z           self.distribution.run_command(command)
2024-07-31T11:31:48.7872369Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/dist.py", line 945, in run_command
2024-07-31T11:31:48.7872472Z           super().run_command(command)
2024-07-31T11:31:48.7872946Z         File "/tmp/pip-build-env-ubj64_59/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
2024-07-31T11:31:48.7873046Z           cmd_obj.run()
2024-07-31T11:31:48.7873148Z         File "<string>", line 243, in run
2024-07-31T11:31:48.7873351Z         File "/usr/lib/python3.12/subprocess.py", line 413, in check_call
2024-07-31T11:31:48.7873492Z           raise CalledProcessError(retcode, cmd)
2024-07-31T11:31:48.7873913Z       subprocess.CalledProcessError: Command '['/usr/bin/cmake', '--build', '.', '--', '-j', '16']' returned non-zero exit status 2.`

like in: https://github.com/openvinotoolkit/openvino/actions/runs/10179302836/job/28154933667

Reply:
ok, if it's a temporary WA for ONNX issue, let's keep gcc-10 for now with comment to update this place once we migrate to newer ONNX